output/cloud/expv2/output.go

```diff
@@ -42,7 +59,32 @@ func (o *Output) SetTestRunStopCallback(stopFunc func(error)) {
 func (o *Output) Start() error {
 	o.logger.Debug("Starting...")
 
-	// TODO: add start implementation
+	var err error
+	o.collector, err = newCollector(
+		&o.SampleBuffer,
+		o.config.AggregationPeriod.TimeDuration(),
+		o.config.AggregationWaitPeriod.TimeDuration())
+	if err != nil {
+		return err
+	}
+	o.metricsFlusher = noopFlusher{
+		referenceID: o.referenceID,
+		bq:          &o.collector.bq,
+	}
+
+	timeframe := o.config.MetricPushInterval.TimeDuration()
+	pf, err := output.NewPeriodicFlusher(timeframe, o.flushMetrics)
+	if err != nil {
+		return err
+	}
+	o.periodicFlusher = pf
+
+	timeframe = o.config.AggregationPeriod.TimeDuration()
+	pfc, err := output.NewPeriodicFlusher(timeframe, o.collector.CollectSamples)
+	if err != nil {
+		return err
+	}
+	o.periodicCollector = pfc
```

Comment:
```suggestion
	o.periodicFlusher, err = output.NewPeriodicFlusher(timeframe, o.flushMetrics)
	if err != nil {
		return err
	}

	timeframe = o.config.AggregationPeriod.TimeDuration()
	o.periodicCollector , err = output.NewPeriodicFlusher(timeframe, o.collector.CollectSamples)
	if err != nil {
		return err
	}
```

To be honest I would even drop the `timeframe` variable jsut so I don't have to wonder if we are using it anywhere else.

Reply:
This still seems to be a problem @codebien maybe you rebased without it?